TruckMuncher/VendorMapViewController.swift

```diff
@@ -66,6 +67,7 @@ class VendorMapViewController: UIViewController, MKMapViewDelegate, CLLocationMa
         if trucks[selectedTruckIndex].isInServingMode {
             centerMapOverCoordinate(CLLocationCoordinate2D(latitude: trucks[selectedTruckIndex].latitude, longitude: trucks[selectedTruckIndex].longitude))
         }
+        menu = RMenu.objectsWhere("truckId = %@", trucks[selectedTruckIndex].id).firstObject() as RMenu
```

Comment:
We should use the `MenuManager.getMenu` method here instead. The only thing would be timing. They could click serving mode before the menu object is returned. If we get it out of the cache, the latest data for out of stock items wouldn't be on the device. Using that, it would also fix the crash in the MenuViewController I saw earlier without any further changes in that class.